c7n/resources/asg.py

```diff
@@ -492,6 +492,44 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('progagated-tags')
+class PropagatedTagFilter(Filter):
+    """
+    This filter is designed to find all autoscalinggroups that have a list
+    of tag keys (provided) that are set to propagate to new instances. Using
+    this will allow for easy validation of asg tag sets are in place across an
+    account for compliance
+
+    - name: asg-propagated-tags
+      resource: asg
+      filters:
+        - type: propagated-tags
+          keys:
+            - 'ABC'
+            - 'BCD'
+    """
+    schema = type_schema(
+        'progagated-tags',
+        keys={'type': 'array', 'items': {'type': 'string'}})
+
+    def process_asg_tags(self, asg):
+        keys = self.data.get('keys', [])
+        tags = asg.get('Tags', [])
+        results = []
+        for tag in tags:
+            if tag['Key'] in keys and tag['PropagateAtLaunch']:
+                results.append(tag['Key'])
+        if len(keys) == len(results):
+            return asg
+
+    def process(self, asgs, event=None):
+        origin = len(asgs)
+        with self.executor_factory(max_workers=3) as w:
+            asgs = [a for a in list(w.map(self.process_asg_tags, asgs)) if a]
```

Comment:
same issue as other pr re batching across the thread / worker boundary to avoid lock contention. 

also filter(None, [1,2,None,4]) = [1,2,4]